frontend/src/api/apiUtils.ts

```diff
@@ -14,13 +20,66 @@ interface IRefreshTokenResponse {
   status: number;
 }
 
+export const getBackendPath = (path?: string) =>
+  `${BACKEND_API_HOST}${VERSION_PREFIX}${path ?? ''}`;
+
 const refreshAccessToken = async (refreshToken: string) =>
   await axios
-    .post(`${BACKEND_API_HOST}${VERSION_PREFIX}${backendServicesPaths.auth.refresh}`, {
+    .post(getBackendPath(backendServicesPaths.auth.refresh), {
       refreshToken: refreshToken,
     })
     .then((resp) => resp as IRefreshTokenResponse);
 
+// Wrapper around axios to make requests to the backend.
+export function requestBackend(requestParams: RequestBackendParams) {
+  requestParams = {
+    requireAuthentication: true,
+    ...requestParams,
+  };
+
+  let config = {};
+
+  // generate axios instance to handle public requests
+  let axiosInstance = axios.create({
+    baseURL: getBackendPath(),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+
+  if (requestParams.requireAuthentication) {
+    const { protectedRequests, protectedRequestConfig } = generateAuthInterceptor(
+      requestParams.authContext,
+    );
+    axiosInstance = protectedRequests;
+    config = protectedRequestConfig;
```

Comment:
Hmm, this still seems off. Usually people use the singleton pattern, since the 'authenticated' axios always has the same config. We can create a singleton for authenticated requests, and one for public endpoints.

We can then remove `generateAuthInterceptor` and directly call the respective axios based on `requestParams.requireAuthentication`